### ISSUE TITLE
Add support for Solid Queue

### ISSUE THREAD

**Issue by luismiramirez** [appsignal/appsignal-ruby#1004]
Solid Queue was announced in Rails World 2023.

Start working on its support once released.

https://rubygems.org/gems/solid_queue

The current 0.1.0 gem present in Rubygems is empty.

## To do

- [x] Instrument jobs
- [x] Report errors from jobs
- [ ] Report _useful_ metadata from jobs: queue name, priority, severity, etc. whatever metadata it has and is useful.
- [ ] Extra: Report metrics about jobs, similar to our Sidekiq instrumentation. Focus first on the instrumentation and then the metrics. These things can be shipped separately if needed.

**Comment by johnnagro** [comment]
The code is out now: https://github.com/basecamp/solid_queue

**Comment by HusseinMorsy** [comment]
@luismiramirez  Any progress on supporting solid queue?

**Comment by luismiramirez** [comment]
Hi @HusseinMorsy we'll work on this in the upcoming future. We'll let you know as soon as we start with it.

**Comment by shairyar** [comment]
https://app.intercom.com/a/inbox/yzor8gyw/inbox/shared/unassigned/conversation/16410700292701?view=List (private link)

**Comment by miharekar** [comment]
Right now the workaround is to add this to your production.rb
```
config.solid_queue.on_thread_error = ->(error) { Appsignal.send_error(error) }
```

**Comment by jeffkreeftmeijer** [comment]
Hey all! I've been doing some testing to see what the status of solid_queue support is, and it's working in [my example app with solid_queue and ActiveJob](https://github.com/appsignal/test-setups/tree/main/ruby/rails7-solid-queue/app). I'm getting both performance and error jobs reported in AppSignal.

@HusseinMorsy @miharekar am I correct that you're not getting performance samples for your jobs, but no errors reported from your apps (without adding your own error handlers)? If so, could you tell me a bit more about your applications? I'm specifically looking to see what the differences are between my example app and your apps.

**Comment by miharekar** [comment]
@jeffkreeftmeijer Hmm didn't work out-of-the-box for me at the time. Let me try again.

**Comment by jeffkreeftmeijer** [comment]
I'm closing this one for now. Feel free to reopen if you're having trouble getting AppSignal to report metrics from SolidQueue.

**Comment by tmaier** [comment]
@jeffkreeftmeijer  if this gem is supported OOTB, please state so at https://docs.appsignal.com/ruby/integrations/active-job.html

**Comment by unflxw** [comment]
Hi @tmaier!

Our ActiveJob integration aims to be compatible with all adapters, including SolidQueue, which we support. We'll look into amending our documentation to make it clearer.
